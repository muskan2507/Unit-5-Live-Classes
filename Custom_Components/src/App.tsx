
import './App.css';
import Accordian from './Components/Accordian';
import OtpInput from './Components/OtpInput';
import Tabs from './Components/Tabs';
const data = [
  {
    desc:
      "A dog is a type of domesticated animal. Known for its loyalty and faithfulness, it can be found as a welcome guest in many households across the world.",
    title: "This is panel header 1",
  },
  {
    desc:
      "A dog is a type of domesticated animal. Known for its loyalty and faithfulness, it can be found as a welcome guest in many households across the world.",
      title: "This is panel header 2",
  },
  {
    desc:
      "A dog is a type of domesticated animal. Known for its loyalty and faithfulness, it can be found as a welcome guest in many households across the world.",
      title: "This is panel header 3",
  },
  {
    desc:
      "A dog is a type of domesticated animal. Known for its loyalty and faithfulness, it can be found as a welcome guest in many households across the world.",
      title: "This is panel header 4",
  },
  {
    desc:
      "A dog is a type of domesticated animal. Known for its loyalty and faithfulness, it can be found as a welcome guest in many households across the world.",
      title: "This is panel header 5",
  },
];
const data1 = [
  {title:"One",desc:"hi one!"},
  {title:"Two",desc:" hello two!"},
  {title:"Three",desc:"sit three!"},
  {title:"Four",desc:"bye four!"},
]
function App() {
  const handleChange=(otp:string)=>{
    console.log(otp)
  }
  return (
    <div className="App">
      {/* -----------------------------------------OTP Box---------------------------------- */}
    <OtpInput inputNumber={5} onChange={handleChange} />
    <br /><hr /><br />
    {/* ---------------------------------------------Accordian-------------------------------- */}
    <Accordian data={data}/>
    <br /><hr /><br />
    {/* --------------------------------------------Tabs----------------------------------------- */}
    <Tabs data={data1}/>
    </div>
  );
}

export default App;
