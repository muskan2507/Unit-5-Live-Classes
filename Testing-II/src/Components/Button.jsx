import React from "react";
import "./btn.css";
const Button = ({ variant, colorScheme, onClick, children }) => {
  return (
    <button
    data-testid="cButton"
      onClick={onClick}
      className={`btn ${colorScheme || "red"} ${variant || "ghost"}`}
    >
      {children}
    </button>
  );
};

export default Button;
