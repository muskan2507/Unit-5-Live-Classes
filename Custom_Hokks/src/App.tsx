import axios from 'axios';
// import { useDebouncedCallback } from '@react-hookz/web';
import { useState } from 'react';
import './App.css';
import useFetch from './hooks/useFetch';
import useTimeout from './hooks/useTimeout';
const URL="https://api.github.com/search/users"
 export type User ={
  id:number;
  login:string;
  avatar_url:string;
}
function App() {
  const toShow=useTimeout(2);
  const [text,setText]=useState("");
  const {data,err,loading}=useFetch(URL,{
    per_page:5,
    q:text || "masai-codes"
  })
   
  return (
    <div className="App">
      {toShow? (<h1 style={{color:'red'}}>Hello</h1>) : (<h1>Waiting for timer..</h1>)}
      <br/><br/><hr/><br/>
      <input onChange={(e)=>setText(e.target.value)}></input>
      
    { loading? (<div>...Loading</div>) : err? (<div>Something Went Wrong</div>): (data.map(User=>{
      return (
        <div>
        {User.login}<br/>
        <img style={{height:"200px",width:"200px"}} src={User.avatar_url}/>
        <br/>
        </div>
      )
    }))}
    </div>
  );
}

export default App;
