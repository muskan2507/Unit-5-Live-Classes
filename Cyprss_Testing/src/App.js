import logo from './logo.svg';
import './App.css';
import { useEffect, useState } from 'react';
import axios from 'axios';

function App() {
  const [count,setCount]=useState(0)
  const get=()=>{
    axios.get("http://localhost:8080/counter").then(({data})=>{
      setCount(data.value)
    })
  }
  useEffect(()=>{
    get()
  },[])
  return (
    <div className="App">
      <header className="App-header">
       
        <h3 className='counter-value'>Count is: {count}</h3>
         <button className='counterBtn' onClick={()=>{
           axios.post("http://localhost:8080/counter",{
             value:count+1
           }).then(()=>{
             setCount(count+1)
           })
         }}>Add Count</button>
      
        
      
      </header>
    </div>
  );
}

export default App;
