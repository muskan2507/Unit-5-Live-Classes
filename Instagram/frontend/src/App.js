import './App.css';
import Login from './Components/Login';
import {BrowserRouter, Route, Routes} from "react-router-dom"
import Feed from './Components/Feed';
function App() {
  return (
    <div className="App">
    <BrowserRouter>
     <Routes>
       <Route path="/login" element={ <Login/>}/>
       <Route path="/feed" element={ <Feed/>}/>
     </Routes>
    </BrowserRouter>
    </div>
  );
}

export default App;
