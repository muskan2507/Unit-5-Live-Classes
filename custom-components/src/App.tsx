
import './App.css';
import OtpInput from './Components/OtpInput';

function App() {
  return (
    <div className="App">
    <OtpInput inputNumber={4}/>
    </div>
  );
}

export default App;
