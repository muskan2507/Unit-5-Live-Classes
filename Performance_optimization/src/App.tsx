import { useMemo, useState } from 'react';
import './App.css';
import Fib from './Components/Fib';
import Todos from './Components/Todos';
import { Counter } from './Components/useReducer';

function App() {
  
  return (
    <div className="App">
    <Fib/>
    <br /><hr /><br />
    <Todos/>
    <br /><hr /><br />
   <Counter/>
   
    </div>
  );
}

export default App;
