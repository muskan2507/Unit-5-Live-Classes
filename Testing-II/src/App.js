import logo from './logo.svg';
import './App.css';
import Button from './Components/Button';
import { useState } from 'react';

function App() {
  const [theme,setTheme]=useState("light")
  return (
    <div className="App">
    <h3>Theme is {theme}</h3>
     <Button onClick={()=>{
       setTheme(theme==="light"?"dark":"light")
     }}>Change theme</Button>
     {/* <Button colorScheme={"green"} variant={"grey"}>Click me</Button> */}
    </div>
  );
}

export default App;
