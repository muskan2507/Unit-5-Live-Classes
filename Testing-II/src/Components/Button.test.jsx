import { fireEvent, render, screen } from "@testing-library/react"
import App from "../App"
import Button from "./Button"
import renderer from "react-test-renderer"

describe("Button Tests",function(){
    it("should have button in dom",function(){
        render(<Button>Click me</Button>)
        let button=screen.getByText("Click me")
        expect(button).toBeInTheDocument();
    })
    it("should render based on testid",function(){
        render(<Button>Click me</Button>)
        let button=screen.getByTestId("cButton")
        expect(button).toBeInTheDocument();
    })
    it("should render on empty button",function(){
        render(<Button></Button>)
        let button=screen.getByTestId("cButton")
        expect(button).toBeEmptyDOMElement();
    })
    it("should change theme",function(){
        render(<App></App>)
        let h3=screen.getByText("Theme is light")
        expect(h3).toHaveTextContent("light");
        let themeButton=screen.getByText("Change theme")
        fireEvent.click(themeButton)
        expect(h3).toHaveTextContent("dark")    
    })
    it("should call given function",function(){
        const mockfn=jest.fn();
        render(<Button onClick={mockfn}>Click me</Button>)
        const btn=screen.getByText("Click me")
        fireEvent.click(btn)
        expect(mockfn).toBeCalled()
    })
    it("should match snapshot",function(){
        const tree=renderer.create(<Button>Add Counter</Button>).toTree()
        expect(tree).toMatchSnapshot()
    })
})