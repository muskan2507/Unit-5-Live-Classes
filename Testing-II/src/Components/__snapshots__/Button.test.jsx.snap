// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests should match snapshot 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "children": "Add Counter",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": "Add Counter",
      "className": "btn red ghost",
      "data-testid": "cButton",
      "onClick": undefined,
    },
    "rendered": Array [
      "Add Counter",
    ],
    "type": "button",
  },
  "type": [Function],
}
`;
